### PR TITLE
Add get item endpoint

### PR DESCRIPTION
**Pull request by samstradling**
Adds the endpoint to get an item on a list:

```http
GET /lists/<list_id>/items/<item_id>
```

### CODE REVIEW

File: db/deleteItem.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package db
```
Comment (alexNaylorTeece) [inline db/deleteItem.go:1]: Are you intending to change this file in this pr?

File: db/db.go [hunk 2]
```diff
@@ -6,6 +6,8 @@ import (
 
 // DB - interface for talking to the database
 type DB interface {
+	DeleteItem(*string, *string) error
```
Comment (alexNaylorTeece) [inline db/db.go:9]: Assume this is meant to be in the different PR?

File: db/deleteItem_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package db
```
Comment (alexNaylorTeece) [inline db/deleteItem_test.go:1]: Is this meant to change in this PR?

**Review by alexNaylorTeece — CHANGES_REQUESTED** [review]
Not sure whether this PR have more changes than you want in it. I'll take a look again once you confirm

**Comment by samstradling** [comment]
> Not sure whether this PR have more changes than you want in it. I'll take a look again once you confirm

These aren't in the PR anymore

File: db/getItem_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+)
+
+func TestGetItem(t *testing.T) {
+	listID := "474c2Fff7"
+	itemID := "b6cf642d"
+	name := "Cheese"
+	tableName := "items-table"
+	tests := []struct {
+		name        string
+		outputErr   error
```
Comment (alexNaylorTeece) [inline db/getItem_test.go:19]: Not totally clear this is for mock at a glance

**Review by alexNaylorTeece — APPROVED** [review]
lgtm
